apps/browser-extension-wallet/src/features/dapp/components/confirm-transaction/VotingProceduresContainer.tsx

```diff
@@ -24,54 +30,69 @@ export const VotingProceduresContainer = (): React.ReactElement => {
     getVotingProcedures();
   }, [request]);
 
+  useEffect(() => {
+    if (!walletState?.transactions.history || userAckNonRegisteredState) return;
+    setIsNonRegisteredUserModalVisible(hasValidDrepRegistration(walletState.transactions.history));
+  }, [walletState?.transactions.history, userAckNonRegisteredState]);
+
   const explorerBaseUrl = useCexplorerBaseUrl();
 
   return (
-    <VotingProcedures
-      dappInfo={dappInfo}
-      data={votingProcedures.map((votingProcedure) => {
-        const voterType = getVoterType(votingProcedure.voter.__typename);
+    <>
+      <NonRegisteredUserModal
+        visible={isNonRegisteredUserModalVisible}
+        onConfirm={() => {
+          setUserAckNonRegisteredState(true);
+          setIsNonRegisteredUserModalVisible(false);
+        }}
+        onClose={() => disallowSignTx(true)}
+      />
+      <VotingProcedures
+        dappInfo={dappInfo}
+        data={votingProcedures.map((votingProcedure) => {
+          const voterType = getVoterType(votingProcedure.voter.__typename);
 
-        const drepId =
-          voterType === VoterTypeEnum.DREP
-            ? drepIDasBech32FromHash(votingProcedure.voter.credential.hash)
-            : votingProcedure.voter.credential.hash.toString();
-        return {
-          voter: {
-            type: t(`core.VotingProcedures.voterTypes.${voterType}`),
-            dRepId: drepId
-          },
-          votes: votingProcedure.votes.map((vote) => ({
-            actionId: {
-              index: vote.actionId.actionIndex,
-              txHash: vote.actionId.id.toString(),
-              txHashUrl: `${explorerBaseUrl}/${vote.actionId.id}`
+          const drepId =
+            voterType === VoterTypeEnum.DREP
+              ? drepIDasBech32FromHash(votingProcedure.voter.credential.hash)
+              : votingProcedure.voter.credential.hash.toString();
+          return {
+            voter: {
+              type: t(`core.VotingProcedures.voterTypes.${voterType}`),
+              dRepId: drepId
             },
-            votingProcedure: {
-              vote: t(`core.VotingProcedures.votes.${getVote(vote.votingProcedure.vote)}`),
-              anchor: !!vote.votingProcedure.anchor?.url && {
-                url: vote.votingProcedure.anchor?.url,
-                hash: vote.votingProcedure.anchor?.dataHash.toString()
+            votes: votingProcedure.votes.map((vote) => ({
+              actionId: {
+                index: vote.actionId.actionIndex,
+                txHash: vote.actionId.id.toString(),
+                txHashUrl: `${explorerBaseUrl}/${vote.actionId.id}`
+              },
+              votingProcedure: {
+                vote: t(`core.VotingProcedures.votes.${getVote(vote.votingProcedure.vote)}`),
+                anchor: !!vote.votingProcedure.anchor?.url && {
```

Comment:
could refactor to 
```suggestion
                anchor: !!vote.votingProcedure.anchor && {
```
which means you don't need to use `?` in the subsequent 2 lines

Reply:
done